core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -513,6 +523,16 @@ public SliceBuilder slice() {
         return sliceBuilder;
     }
 
+
+    public CollapseBuilder collapse() {
```

Comment:
Looking at the latest change, and I don't see that the `collapse` field is taken into account here and the no changes have been made to `SearchSourceBuilderTests.java`. Can you make these changes?